lib/protoparser/opentelemetry/firehose/parser.go

```diff
@@ -32,8 +49,18 @@ func ProcessRequestBody(b []byte) ([]byte, error) {
 
 	var dst []byte
 	for _, r := range req.Records {
-		dst = append(dst, r.Data...)
+		for len(r.Data) > 0 {
```

Comment:
This logic looks really weird. So, the `r.Data` contains a series of [ResourceMetrics](https://github.com/open-telemetry/opentelemetry-proto/blob/34d29fe5ad4689b5db0259d3750de2bfa195bc85/opentelemetry/proto/metrics/v1/metrics.proto#L48) protobuf messages pre-pended with their length? :0

It would be great adding tests for these cases to both `firehose` package and to `opentelemetry` package with the real-world JSON messages obtained from Amazon Firehose.

Reply:
yes, prometheus protobuf exposition is implemented in the same manner

Reply:
@valyala pushed sample generated by Firehose